### PR TITLE
Replace Legacy Request Data in Venues Admin

### PR DESCRIPTION
**Pull request by tn3rb**
like some previous PRs, this PR updates the Venues Admin by replacing all uses of the legacy _req_data array with functionality on the new Request class

### CODE REVIEW

**Review by knazart — APPROVED** [review]
Just one formatting issue. Otherwise looks good to me.

**Comment by garthkoyle** [comment]
I ran through the Venue requests and didn't run into any issues.

**Comment by garthkoyle** [comment]
Tony reported that if you add html to the Venue Category (Venue > Categories > Edit), then save the page the HTML is stripped.

**Comment by tn3rb** [comment]
pushed a fix for the HTML getting stripped

File: core/services/request/sanitizers/AllowedTags.php [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+     * @return array[]
+     */
+    public static function getAllowedTags()
+    {
+        if (empty(AllowedTags::$allowed_tags)) {
+            AllowedTags::initializeAllowedTags();
+        }
+        return AllowedTags::$allowed_tags;
+    }
+}
```
Comment (knazart) [inline core/services/request/sanitizers/AllowedTags.php]: No new line.

File: admin_pages/messages/Messages_Admin_Page.core.php [hunk 2]
```diff
@@ -2973,18 +2947,45 @@ protected function _insert_or_update_message_template($new = false)
...
+        }
+        // remove slashes so wp_kses() works properly
+        // wp_kses_stripslashes() only removes slashes from double-quotes,
+        // so attributes using single quotes always appear invalid.
+        $content = stripslashes($content);
+        $content = wp_kses($content, wp_kses_allowed_html('post'));
+        // But currently the models expect slashed data, so after wp_kses()
+        // runs we need to re-slash the data. Sheesh.
+        // See https://events.codebasehq.com/projects/event-espresso/tickets/11211#update-47321587
+        return addslashes($content);
```
Comment (knazart) [inline admin_pages/messages/Messages_Admin_Page.core.php:2984]: 🤨 😅

**Review by knazart — APPROVED** [review]

**Comment by garthkoyle** [comment]
HTML is not stripped from the category description anymore, but, I think Master needs to be merged into this branch and then I can re-test.

**Comment by garthkoyle** [comment]
Retesting after rebasing, the HTML persists after updating the category description.

**Comment by Pebblo** [comment]
Retested this and all seems fine.

Tested adding/editing a venue
Trashing/restoring venue.
Trashing then Perm Delete.
Google maps settings saved.

Categories work fine.
Category description now saves HTML as expected.

File: admin_pages/messages/Messages_Admin_Page.core.php (hunk 2 above)
Comment (tn3rb) [reply admin_pages/messages/Messages_Admin_Page.core.php:3096]: pretty sure that's Darren's comments and code

**Pull request merged**